binary_search_tree/binary_search_tree.py

```diff
@@ -78,46 +82,102 @@ def for_each(self, cb):
         if self.right:
             self.right.for_each(cb)
 
+    #iterative solution DFT:
+        stack = Stack()
+        stack.push(self)
+
+        while stack.len() > 0:
+            current = stack.pop()
+            if current.left:
+                stack.push(current.left)
+            if current.right:
+                stack.push(current.right)
+            cb(current.value)
+
+
     # DAY 2 Project -----------------------
 
     # Print all the values in order from low to high
     # Hint:  Use a recursive, depth first traversal
     def in_order_print(self, node):
-        if self.value is not None:
-            if self.left:
-                self.left.in_order_print(self.left)
-            if self.right:
-                self.right.in_order_print(self.right)
-        else:
-            return None
+        if not node:
+            return
+        self.in_order_print(node.left)
+        print(node.value)
+        self.in_order_print(node.right)
     # Print the value of every node, starting with the given node,
     # in an iterative breadth first traversal
     def bft_print(self, node):
+        #FIFO OR LILO
+        #go by layer
+        '''
+        STEPS
+        MAKE QUEUE
+        PUT ROOT IN QUEUE
+        WHILE  queue is not empty
+            pop out front of queue
+            cb print node.value
+            if left:
+                add left to back of queue
+            if right:
+                add right to back of queue
+        '''
         queue = Queue()
-        if node:
-            queue.enqueue(node.value)
-        while node:
+        queue.enqueue(node)
+        while  queue.len() > 0:
+            node = queue.dequeue()
+            print(node.value)
             if node.left:
                 queue.enqueue(node.left)
             if node.right:
                 queue.enqueue(node.right)
-            queue.dequeue(0)
-            if not queue:
-                break
-            node = queue[0]
 
     # Print the value of every node, starting with the given node,
     # in an iterative depth first traversal
     def dft_print(self, node):
+        '''
+        #STEPS
+        make a stack
+        put root in stack
+        pop root out of stack
+        while stack not empty
+            pop root out of stack
+            print(node.value)
+            if left
+                add left to stack
+            if right:
+                add right to stack
+        '''
+        #FILO OR LIFO
         stack = Stack()
+        stack.push(node)
+        while  stack.len() > 0:
+            node = stack.pop()
+            print(node.value)
+            if node.left:
+                stack.push(node.left)
+            if node.right:
+                stack.push(node.right)
 
     # STRETCH Goals -------------------------
     # Note: Research may be required
 
     # Print In-order recursive DFT
     def pre_order_dft(self, node):
```

Comment:
Let's do some quick research, make sure that that is all pre-order and post-order requires!!!